adoption of Cloud SDK 5 in CAP Java (#627)
cap-js/docs: java/migration.md

<!-- TODO: Move this to "Development" section -->

<span id="afterenablingodata" />

## Cloud SDK 4 to 5 { #cloudsdk5 }

> CAP Java `2.6.0` and higher is compatible with Cloud SDK in version 4 and 5. For reasons of backward compatibility, CAP Java assumes Cloud SDK 4 as the default. However, we highly recommended to use at least version `5.2.0` of Cloud SD. To upgrade your CAP Java application to Cloud SDK 5, add the following maven dependency to your CAP Java application:

Reviewer: ```suggestion
CAP Java `2.6.0` and higher is compatible with Cloud SDK in version 4 and 5. For reasons of backward compatibility, CAP Java assumes Cloud SDK 4 as the default. However, we highly recommend to use at least version `5.2.0` of Cloud SDK. To upgrade your CAP Java application to Cloud SDK 5 you don't need to do anything if you're using the Cloud SDK integration package (`cds-integration-cloud-sdk`). If you're using Cloud SDK directly, add the following maven dependency to your CAP Java application:
```

Reviewer: Is that statement correct?

Author: `... ,in most cases, you  don't need to adapt any code if you rely on the Cloud SDK integration package (`cds-integration-cloud-sdk`). In these cases, it is sufficient to add the following maven dependency to your CAP Java application:`

You still have to add the dependency. If you rely on our integration package, there will be most likely no code change required.

```xml
<dependency>
	<groupId>com.sap.cloud.sdk.cloudplatform</groupId>
	<artifactId>connectivity-apache-httpclient4</artifactId>
</dependency>
```

> In case your application uses the Cloud SDK integration package (`cds-integration-cloud-sdk`), no code changes in your custom code should be necessary.

Reviewer: ```suggestion
```

<!-- TODO: Move this to "Development" section -->

<span id="afterenablingodata" />

## Cloud SDK 4 to 5 { #cloudsdk5 }

> CAP Java `2.6.0` and higher is compatible with Cloud SDK in version 4 and 5. For reasons of backward compatibility, CAP Java assumes Cloud SDK 4 as the default. If you want to upgrade your CAP Java application to Cloud SDK 5, it is necessary to add the following maven dependency to your CAP Java application:

Reviewer: Why do we need to add exactly this dependency?
```suggestion
CAP Java `2.6.0` and higher is compatible with Cloud SDK in version 4 and 5. For reasons of backward compatibility, CAP Java assumes Cloud SDK 4 as the default. However, we highly recommended to use at least version `5.2.0` of Cloud SD. To upgrade your CAP Java application to Cloud SDK 5, add the following maven dependency to your CAP Java application:
```

```xml
<dependency>
	<groupId>com.sap.cloud.sdk.cloudplatform</groupId>
	<artifactId>connectivity-apache-httpclient4</artifactId>
</dependency>
```

> In case your application makes use of Cloud SDK integration (`cds-integration-cloud-sdk`), no code changes in custom code should be necessary.

Reviewer: Why do we mention those two packages explicitly? Are those heavily used by our stakeholders?
```suggestion
In case your application uses the Cloud SDK integration package (`cds-integration-cloud-sdk`), no code changes in your custom code should be necessary.
```

Author: It is actually only about one. It´s just the human readable name and its technical maven module name. We mention it here explicitly, it provides convenience features for integrating with Cloud SDK (which is an external dependency for us). This integration part is implemented by us and hence under our control. However, applications can also use Cloud SDK directly which means that they can also use APIs which have changed incompatible in Cloud SDK.

> It is highly recommended to use at least version `5.2.0` of Cloud SDK.

Reviewer: ```suggestion
```
